src/connector/src/parser/postgres.rs

```diff
@@ -464,3 +520,112 @@ fn pg_numeric_to_string(val: Option<PgNumeric>) -> Option<String> {
         None
     }
 }
+
+#[derive(Clone, Debug)]
+struct EnumParser {
+    value: String,
+}
+
+impl<'a> FromSql<'a> for EnumParser {
+    fn from_sql(
+        _ty: &Type,
+        raw: &'a [u8],
+    ) -> Result<Self, Box<dyn std::error::Error + 'static + Sync + Send>> {
+        Ok(EnumParser {
+            value: String::from_utf8_lossy(raw).into_owned(),
+        })
+    }
+
+    fn accepts(ty: &Type) -> bool {
+        matches!(ty.kind(), Kind::Enum(_))
+    }
+}
+
+impl ToSql for EnumParser {
+    to_sql_checked!();
+
+    fn accepts(ty: &Type) -> bool {
+        matches!(ty.kind(), Kind::Enum(_))
+    }
+
+    fn to_sql(
+        &self,
+        ty: &Type,
+        out: &mut BytesMut,
+    ) -> Result<IsNull, Box<dyn std::error::Error + Sync + Send>>
+    where
+        Self: Sized,
+    {
+        match ty.kind() {
+            Kind::Enum(e) => {
+                if e.contains(&self.value) {
+                    out.extend_from_slice(self.value.as_bytes());
+                    Ok(IsNull::No)
+                } else {
+                    Err(format!(
+                        "EnumParser value {} is not in the enum type {:?}",
+                        self.value, e
+                    )
+                    .into())
+                }
+            }
+            _ => Err("EnumParser can only be used with ENUM types".into()),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use postgres::{Client, NoTls};
+
+    use crate::parser::postgres::EnumParser;
+
+    const DB: &str = "postgres";
+    const USER: &str = "kexiang";
+
+    #[ignore]
+    #[test]
+    fn enum_parser_integration_test() {
```

Comment:
Is it possible to do some unit test instead of integration test here? I guess e2e test has already played the integration role? Should have automated test here.

Reply:
In this test, I want to make sure two things:

1. `EnumParser` correctly understood the data in pg.
2.  The data transformed from `EnumParser` can be correctly understood by pg.

So the test cannot run without pg. Here, as I have marked as `#[ignore]`, I just want to provide a simple tool to test `EnumParser`. `e2e test` can already verify its correctness.